geode-connectors/src/distributedTest/java/org/apache/geode/connectors/jdbc/internal/cli/CreateDataSourceCommandDUnitTest.java

```diff
@@ -99,6 +102,45 @@ public void testCreateDataSource() throws Exception {
     verifyThatNonExistentClassCausesGfshToError();
   }
 
+  @Test
+  public void testCreateDataSourceWithJarOptionDoesNotThrowDriverError() {
+    String URL = "jdbc:mysql://localhost/";
+    IgnoredException.addIgnoredException(
+        "No suitable driver");
+    IgnoredException.addIgnoredException(
+        "create data-source failed");
+
+    // aquire the jar to be used
+    final String jdbcJarName = "mysql-connector-java-8.0.15.jar";
+    File mySqlDriverFile = loadTestResource("/" + jdbcJarName);
+    assertThat(mySqlDriverFile).exists();
+    String jarFile = mySqlDriverFile.getAbsolutePath();
+
+    // attempt to create the data source without a deployed jar or a --driver-jar
+    gfsh.executeAndAssertThat(
+        "create data-source --name=mySqlDataSource --username=mySqlUser --password=mySqlPass --pooled=false --url=\""
+            + URL + "\"")
+        .statusIsError();
+
+    gfsh.executeAndAssertThat("deploy --jar=" + jarFile).statusIsSuccess();
+
+    IgnoredException.removeAllExpectedExceptions();
+    IgnoredException.addIgnoredException(
+        "create data-source failed");
+    IgnoredException.addIgnoredException(
+        "com.mysql.cj.jdbc.exceptions.CommunicationsException: Communications link failure");
+    IgnoredException.addIgnoredException(
+        "Access denied for user 'mySqlUser'@'localhost'");
```

Comment:
I don't think it will throw exception like this. Because there is no mysql server running.

Reply:
This exception is added in case the test is run on a developers machine which has a running mysql server as it was weird when the test passed on some machines but didn't on others. On concourse, this statement is not necessary.